### ISSUE TITLE
IMAP Search Negation not working

### ISSUE THREAD

**Issue by anshulCoder** [mscdex/node-imap#698]
Hi,
I was trying a simple search in mails inbox folder here is my code so far:

`imap.search([ ['FROM', 'Pinterest'], 'OR', ['!SUBJECT', 'Sorry. We had to delete one of your Pins on Pinterest.'] ], (err, ids) => {`

Here even after putting a ! sign to negate the condition it was giving a same result like true condition, is the above code correct?

**Comment by mscdex** [comment]
No, the criteria/operand goes first, so it would be something like:

```js
[
  [ 'OR',
    ['FROM', 'Pinterest'],
    ['!SUBJECT', 'Sorry. We had to delete one of your Pins on Pinterest.']
  ]
]
```

assuming you want either of those two to be true.
